tools/serverpod_cli/lib/src/analyzer/dart/endpoints_analyzer.dart

```diff
@@ -36,6 +27,41 @@ class EndpointsAnalyzer {
           resourceProvider: PhysicalResourceProvider.INSTANCE,
         );
 
+  /// Analyze all files in the [AnalysisContextCollection].
+  /// Use [changedFiles] to mark files, that need reloading.
+  Future<List<EndpointDefinition>> analyze({
+    required CodeAnalysisCollector collector,
+    Set<String>? changedFiles,
+  }) async {
+    await _refreshContextForFiles(changedFiles);
+
+    var endpointDefs = <EndpointDefinition>[];
+    for (var context in collection.contexts) {
+      var analyzedFiles = context.contextRoot.analyzedFiles().toList();
+      analyzedFiles.sort();
+      var analyzedDartFiles =
+          analyzedFiles.where((path) => path.endsWith('.dart'));
```

Comment:
Suggestion: Exluce files ending in `_test.dart` just be guard against potential issues if users put their test files in the same folder as the endpoint.

Reply:
As this is a new feature (opposed to this refactor) I will implement it as a separate PR along with tests.